Add Java synthetic property references (#3504)
JetBrains/kotlin-web-site: docs/topics/jvm/java-interop.md

## Getters and setters

Methods that follow the Java conventions for getters and setters (no-argument methods with names starting with `get`
and single-argument methods with names starting with `set`) are represented as properties in Kotlin. Such properties are 

> also called _synthetic_. 

Reviewer: I think it makes sense to mention the whole "synthetic properties" phrase in the definition to make it more clear

}
```

> `calendar.firstDayOfWeek` above is a synthetic property.

Reviewer 2: ```suggestion
`calendar.firstDayOfWeek` above is an example of a synthetic property.
```

Note that, if the Java class only has a setter, it isn't visible as a property in Kotlin because Kotlin doesn't support
set-only properties.

## Java synthetic property references

> This feature is [Experimental](components-stability.md#stability-levels-explained). It may be dropped or changed at any time.

> > Use it only for evaluation purposes. We'd appreciate your feedback on it in [YouTrack](https://kotl.in/issue).

Reviewer 2: ```suggestion
> We recommend that you use it only for evaluation purposes.
```
Danil asked me to not request feedback via YouTrack on experimental features in Docs. Only in what's new. So I guess the same applies here :)

>
{type="warning"}

Starting from Kotlin 1.8.20, you can create references to Java synthetic properties. For example, for such Java code:

```
public class Person {
    private String name;
    private int age;

    public Person(String name, int age) {
        this.name = name;
        this.age = age;
    }

    public String getName() {
        return name;
    }

    public int getAge() {
        return age;
    }
}
```

Kotlin has always allowed you to write `person.age`, where `age` is a synthetic property. Now, you can also 
create references to `Person::age` and `person::age`. All the same works for `name`, as well.

```
val persons = listOf(Person("Jack", 11), Person("Sofie", 12), Person("Peter", 11))
    Persons
         // Call a reference to Java synthetic property:
        .sortedBy(Person::age)
         // Call Java getter via the Kotlin property syntax:
        .forEach { person -> println(person.name) }
}
```

### How to enable Java synthetic property references {initial-collapse-state="collapsed"}

> To try this feature out, enable the `-language-version 1.9` compiler option. In a Gradle project, you can do so 

Reviewer 2: ```suggestion
To enable this feature, set the `-language-version 1.9` compiler option. In a Gradle project, you can do so 
```
